stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ViewDescriptor.java

```diff
@@ -10,19 +10,132 @@
 package com.facebook.stetho.inspector.elements.android;
 
 import android.view.View;
+import android.view.ViewDebug;
+import com.facebook.stetho.common.LogUtil;
 import com.facebook.stetho.common.StringUtil;
 import com.facebook.stetho.common.android.ResourcesUtil;
-import com.facebook.stetho.inspector.elements.AttributeAccumulator;
 import com.facebook.stetho.inspector.elements.AbstractChainedDescriptor;
+import com.facebook.stetho.inspector.elements.AttributeAccumulator;
+import com.facebook.stetho.inspector.elements.StyleAccumulator;
 
 import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
 import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
 
 final class ViewDescriptor extends AbstractChainedDescriptor<View> implements HighlightableDescriptor {
   private static final String ID_ATTRIBUTE_NAME = "id";
-
+  private static final Pattern CAPITAL_PATTERN = Pattern.compile("(?=\\p{Upper})");
   private final MethodInvoker mMethodInvoker;
 
+  /**
+   * NOTE: Only access this via {@link #getViewGetters}.
+   */
+  @GuardedBy("this")
+  private Set<Method> VIEW_GETTERS;
+
+  private final Set<Method> getViewGetters() {
+    if (VIEW_GETTERS == null) {
+      synchronized (this) {
+        Set<String> allowedGetters = new HashSet<>(
```

Comment:
Don't forget to check `VIEW_GETTERS` for `null` again.

https://en.wikipedia.org/wiki/Double-checked_locking